<launch>
  <arg name="dataset_name"   value="TESSE"/>
  <arg name="cache_time"     default="144.0"/>
  <arg name="path_to_vocab"
       default="$(find KimeraVIO)/vocabulary/ORBvoc.yml"/>

  <arg name="body_frame_id"   value="base_link"/>
  <arg name="world_frame_id"  value="world"/>
  <arg name="map_frame_id"    value="map"/>
  <arg name="log_output"      default="false" />
  <arg name="visualize"       default="false"/>
  <arg name="error_to_scale"  default="false"/>

  <arg name="gnd_truth_topic"       value="/tesse/gt"/>
  <arg name="left_cam_topic"        value="/tesse/left_cam"/>
  <arg name="right_cam_topic"       value="/tesse/right_cam"/>
  <arg name="lcd_input_topic"       value="lcd_input"/>
  <arg name="closure_result_topic"  value="result"/>
  <arg name="closure_image_topic"   value="image"/>
  <arg name="error_topic"           value="error"/>
  <arg name="markers_topic"         value="marker_array"/>
  <arg name="poses_topic"           value="pose_array"/>

  <include file="$(find kimera_lcd_ros)/launch/lcd_ros.launch"
    pass_all_args="true"/>

  <!-- Launch Transfrom Converter Node -->
  <node ns="lcd" name="transform_converter_node" pkg="kimera_lcd_ros"
        type="transform_converter_node.py" output="screen">

    <param name="gnd_truth_topic" value="$(arg gnd_truth_topic)"/>
    <param name="body_frame_id" value="$(arg body_frame_id)"/>
    <param name="world_frame_id" value="$(arg world_frame_id)"/>
  </node>

  <!-- Launch RVIZ -->
  <group if="$(arg visualize)">
    <node name="rviz" pkg="rviz" type="rviz" output="screen"
          args="-d $(find lcd_ros)/rviz/lcd_ros_config.rviz">

      <!-- Remap Subscriber Topics -->
      <remap from="/cam0/image_raw"   to="$(arg left_cam_topic)"/>
      <remap from="/cam1/image_raw"   to="$(arg right_cam_topic)"/>
      <remap from="/lcd/result"       to="/lcd/$(arg closure_result_topic)"/>
      <remap from="/lcd/image"        to="/lcd/$(arg closure_image_topic)"/>
      <remap from="/lcd/marker_array" to="/lcd/$(arg markers_topic)"/>
      <remap from="/lcd/pose_array"   to="/lcd/$(arg poses_topic)"/>
    </node>
  </group>
</launch>
